apps/docs/content/docs/editor.mdx

```diff
@@ -694,6 +694,36 @@ editor.run(
 )
 ```
 
+### Hiding shapes
+
+You can pass a `isShapeHidden` predicate when configuring tldraw. This predicate will be called for each shape to determine if it should be hidden.
+
+Hidden shapes will still be present in the store, but
+
+- They will not be shown on the canvas
+- They will not be included in hit test results via `editor.getShapeAtPoint` and `editor.getShapesAtPoint`
+
+Otherwise, they will behave as normal. If you need to restrict certain interactions with hidden shapes, e.g. preventing the user from having a hidden shape selected, you need to handle that in your own code.
+
+e.g.
+
+```tsx
+<Tldraw
+	isShapeHidden={(shape) => shape.meta.hidden}
+	onMount={(editor) => {
+		return react('clean up selection', () => {
+			const selectedShapeIds = editor.getSelectedShapeIds()
+			const filteredSelectedShapeIds = selectedShapeIds.filter((id) => !editor.isShapeHidden(id))
+			if (selectedShapeIds.length !== filteredSelectedShapeIds.length) {
+				editor.setSelectedShapes(filteredSelectedShapeIds)
+			}
+		})
```

Comment:
intrigued on this part too, to me this feels more like the inner workings of tldraw rather than something the user should be controlling, is this something that could be included as part of mounting by default without needing the user to specify? I'd imagine so, as `isShapeHidden` should always return false if the user hasn't set up the `isShapeHidden` function on `Tldraw`.

if the user needs to specify it it might be worth adding a comment to clarify exactly what's going on.

Reply:
We can't add this as builtin behaviour because it's not always what you'd want. E.g. with a layer panel it's entirely possible to select hidden shapes in the panel, and to delete them and so forth. We can't anticipate all the ways that the 'hiddenness' of a shape will inform how different interactions work.

Good call on the comment I'll add one.

Reply:
ah great, that makes sense! I assume exposing a `hiddenShapesSelectable` or `preventSelectingHiddenShapes` that enables/disables this wouldn't work as it's more complex than just that in some instances?

just making sure the surface level API can't be simplified, all good if there's good reasons.